test/unit/math/mix/fun/asinh_test.cpp

```diff
@@ -17,3 +17,28 @@ TEST(mathMixFun, asinh) {
     }
   }
 }
+
+TEST(mathMixMatFun, asinh_varmat) {
+  auto f = [](const auto& x1) {
+    using stan::math::asinh;
+    return asinh(x1);
+  };
+  auto com_args = stan::test::internal::common_args();
```

Comment:
I think all these tests should be for varmat with vector, row vector and Matrix vector arguments.

How about:

```cpp
expect_ad_common_vector_matvar(f);
```

And it does:
```cpp
expect_ad_vector_matvar(f, stan::math::to_vector(stan::test::internal::common_args()));
  // not sure if `common_args` is std::vector or not
```

And that function is something like:

```cpp
expect_ad_vector_matvar(F& f, const auto& x) {
  Eigen::VectorXd v = x;
  Eigen::RowVectorXd r = x.transpose();
  Eigen::MatrixXd m(x.size(), 2);
  com_args_m.block(0, 0, com_args_v.size(), 1) = x;
  com_args_m.block(0, 1, com_args_v.size(), 1) = x.reverse(); // reverse just to mix stuff up
  
  expect_ad_matvar(f, v);
  expect_ad_matvar(f, r);
  expect_ad_matvar(f, m);

  std::vector<Eigen::VectorXd> vv = { v, v };
  std::vector<Eigen::RowVectorXd> rv = { r, r };
  std::vector<Eigen::MatrixXd> mv = { m, m };

  expect_ad_matvar(f, vv);
  expect_ad_matvar(f, rv);
  expect_ad_matvar(f, mv);
}
```

Also does `std::vector<std::vector<...>>` work? Don't think it needs tested I'm just curious.

Reply:
Yeah I think it makes sense to test matrix vector row vector at least to make sure they compile. I'll write something like this up